share/gpodder/extensions/youtube-dl.py

```diff
@@ -577,6 +581,42 @@ def toggle_embed_subtitles(self, widget):
         else:
             self.container.config.embed_subtitles = False
 
+    def check_for_update(self, widget):
+        success = False
+        try:
+            output = subprocess.check_output([sys.executable, '-m', 'pip', 'index', 'versions', program_name]).decode('utf-8')
+            logger.debug(f"pip index version {program_name}: {output}")
+            match = re.search(r'LATEST:\s*(\S*)', output)
+            if match:
+                self.latest_version = match.group(1)
+                success = True
+            else:
+                logger.Error("Could not find LATEST version in pip output")
+                self.latest_version = None
+        except Exception as e:
+            logger.Error(e)
+            self.latest_version = None
+
+        if success:
+            # Need to remove zeros from single digit days and months
+            # Because pip returns versions like 2021.2.9 but the module reports it like 2021.02.09
+            version = ".".join([str(int(i)) for i in youtube_dl.version.__version__.split('.')])
+            if self.latest_version != version:
+                self.update_btn.set_sensitive(True)
+                self.update_label.set_text('%s %s' % (_('Available version:'), self.latest_version))
+            else:
+                self.update_btn.set_sensitive(False)
+                self.update_label.set_text(_('No update available'))
+        else:
+            self.update_btn.set_sensitive(False)
+            self.update_label.set_text(_('Error checking for update'))
+
+    def do_update(self, widget):
+        try:
+            subprocess.check_call([sys.executable, '-m', 'pip', 'install', '--upgrade', program_name])
+        except Exception as e:
+            logger.Error(e)
```

Comment:
I'd capture the output for exception logging:

```suggestion
            subprocess.check_output([sys.executable, '-m', 'pip', 'install', '--upgrade', program_name], stderr=subprocess.STDOUT, encoding='utf-8', timeout=120)
        except subprocess.CalledProcessError as e:
            logger.error("Error running %s: exit code %s, output: %s", e.cmd, e.returncode, e.output)
        except Exception as e:
            logger.exception("Error updating %s: %r", program_name, e)
```

Reply:
It was planned to use a progress dialog and show error output in a dialog, this is a rough draft. Logging the update and error would be a good idea though.